pkg/util/common/common_test.go

```diff
@@ -128,5 +131,27 @@ func TestStructToMap(t *testing.T) {
 				},
 			},
 		},
-	}, StructToMap(top))
+	}
+
+	topNilOutput := map[string]interface{}{
+		"name":  "",
+		"value": 0,
+		"mymap": map[string]interface{}{},
+	}
+
+	cases := []struct {
+		name string
+		in   Top
+		out  map[string]interface{}
+	}{
+		{"notNil", top, topOutput},
+		{"nil", topNil, topNilOutput},
```

Comment:
So this `topNil` is not `nil` actually, just an instance with default values. You need to pass actual `nil` here.

Reply:
Whats the right way to pass in nil here? 
Looking at the code in common.go, do we want to hit the first conditional:
```
	rt, rv := reflect.TypeOf(obj), reflect.ValueOf(obj)
	if rt != nil && rt.Kind() != reflect.Struct {
		return make(map[string]interface{}, 0)
	}
```
The only instances I can get to show a <nil> type are error and an empty interface or an interface where both the type and value are nil.  However, these throw errors when passed into the original func StructToMap.  I've also passed in a nil pointers , which evaluate as expected, but I don't think they satisfy being nil here either.

Reply:
That's the point, if you see `StructToMap` panic on a particular value passed in, it's a bug that needs to be fixed. It should be resilient to these types of inputs.